mscorlib/system/runtime/serialization/safeserializationmanager.cs

```diff
@@ -415,7 +415,11 @@ object IObjectReference.GetRealObject(StreamingContext context)
                 currentType = inheritanceChain.Pop() as RuntimeType;
                 serializationCtor = currentType.GetSerializationCtor();
             }
+#if MONO
```

Comment:
This is not needed. Fix RuntimeConstructorInfo to return false in IsSecurityCritical